great_expectations/expectations/core/expect_column_values_to_be_between.py

```diff
@@ -125,51 +125,7 @@ def validate_configuration(self, configuration: Optional[ExpectationConfiguratio
 
         # Setting up a configuration
         super().validate_configuration(configuration)
-        if configuration is None:
-            configuration = self.configuration
-
-        min_val = None
-        max_val = None
-
-        # Setting these values if they are available
-        if "min_value" in configuration.kwargs:
-            min_val = configuration.kwargs["min_value"]
-
-        if "max_value" in configuration.kwargs:
-            max_val = configuration.kwargs["max_value"]
-
-        try:
-            # Ensuring Proper interval has been provided
-            assert (
-                min_val is not None or max_val is not None
```

Comment:
I couldn't get this assertion to work for `tests/test_definitions/test_expectations_cfe.py`. There're several cases where the test runner thinks that `min_val` and `max_val` are both None.

Reply:
Yes, adding this assertion still fails on my end. Note that this wasn't in the original configuration validation function before -- just in this specific expectation (which I've refactored).

This is what I get

```
=============================================== short test summary info ================================================
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_stdev_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_stdev_to_be_between:positive_zero_stdev_exact_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_mean_to_be_between:type_mismatch_null_observed_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_stdev_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_stdev_to_be_between:positive_zero_stdev_exact_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_mean_to_be_between:type_mismatch_null_observed_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
========================= 54 failed, 1259 passed, 483 skipped, 5 warnings in 147.71s (0:02:27) =========================
```

Example exception:
```
______ test_case_runner_cfe[postgresql/other_expectations/expect_table_column_count_to_be_between:vacuously_true] ______

self = <great_expectations.expectations.core.expect_table_column_count_to_be_between.ExpectTableColumnCountToBeBetween object at 0x7f89e8e04550>
configuration = {"kwargs": {"result_format": "COMPLETE", "include_config": false}, "meta": {}, "expectation_type": "expect_table_column_count_to_be_between"}

    def validate_metric_value_between_configuration(
        self, configuration: Optional[ExpectationConfiguration]
    ):
        # Validating that Minimum and Maximum values are of the proper format and type
        min_val = None
        max_val = None
        parse_strings_as_datetimes = None

        if "min_value" in configuration.kwargs:
            min_val = configuration.kwargs["min_value"]

        if "max_value" in configuration.kwargs:
            max_val = configuration.kwargs["max_value"]

        if "parse_strings_as_datetimes" in configuration.kwargs:
            parse_strings_as_datetimes = configuration.kwargs[
                "parse_strings_as_datetimes"
            ]

        try:
            # Ensuring Proper interval has been provided
            assert any(
                [min_val, max_val]
>           ), "min_value and max_value cannot both be None"
E           AssertionError: min_value and max_value cannot both be None

great_expectations/expectations/expectation.py:1093: AssertionError

During handling of the above exception, another exception occurred:

test_case = {'expectation_type': 'expect_table_column_count_to_be_between', 'skip': False, 'test': {'exact_match_out': False, 'in'... 'vacuously_true'}, 'validator_with_data': <great_expectations.validator.validator.Validator object at 0x7f89e44833d0>}

    def test_case_runner_cfe(test_case):
        if test_case["skip"]:
            pytest.skip()

        # Note: this should never be done in practice, but we are wiping expectations to reuse batches during testing.
        # test_case["batch"]._initialize_expectations()

        evaluate_json_test_cfe(
            validator=test_case["validator_with_data"],
            expectation_type=test_case["expectation_type"],
>           test=test_case["test"],
        )

tests/test_definitions/test_expectations_cfe.py:299:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_utils.py:1208: in evaluate_json_test_cfe
    result = getattr(validator, expectation_type)(**runtime_kwargs)
great_expectations/validator/validator.py:263: in inst_expectation
    raise err
great_expectations/validator/validator.py:213: in inst_expectation
    expectation = expectation_impl(configuration)
great_expectations/expectations/expectation.py:137: in __init__
    self.validate_configuration(configuration)
great_expectations/expectations/core/expect_table_column_count_to_be_between.py:107: in validate_configuration
    self.validate_metric_value_between_configuration(configuration=configuration)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <great_expectations.expectations.core.expect_table_column_count_to_be_between.ExpectTableColumnCountToBeBetween object at 0x7f89e8e04550>
configuration = {"kwargs": {"result_format": "COMPLETE", "include_config": false}, "meta": {}, "expectation_type": "expect_table_column_count_to_be_between"}

    def validate_metric_value_between_configuration(
        self, configuration: Optional[ExpectationConfiguration]
    ):
        # Validating that Minimum and Maximum values are of the proper format and type
        min_val = None
        max_val = None
        parse_strings_as_datetimes = None

        if "min_value" in configuration.kwargs:
            min_val = configuration.kwargs["min_value"]

        if "max_value" in configuration.kwargs:
            max_val = configuration.kwargs["max_value"]

        if "parse_strings_as_datetimes" in configuration.kwargs:
            parse_strings_as_datetimes = configuration.kwargs[
                "parse_strings_as_datetimes"
            ]

        try:
            # Ensuring Proper interval has been provided
            assert any(
                [min_val, max_val]
            ), "min_value and max_value cannot both be None"
            if parse_strings_as_datetimes:
                assert min_val is None or is_parseable_date(
                    min_val
                ), "Provided min threshold must be a dateutil-parseable date"
                assert (
                    max_val is None or is_parseable_date(max_val),
                ), "Provided max threshold must be a dateutil-parseable date"
            else:
                assert min_val is None or isinstance(
                    min_val, (float, int, dict)
                ), "Provided min threshold must be a number"
                if isinstance(min_val, dict):
                    assert (
                        "$PARAMETER" in min_val
                    ), 'Evaluation Parameter dict for min_value kwarg must have "$PARAMETER" key'

                assert max_val is None or isinstance(
                    max_val, (float, int, dict)
                ), "Provided max threshold must be a number"
                if isinstance(max_val, dict):
                    assert "$PARAMETER" in max_val, (
                        "Evaluation Parameter dict for max_value "
                        "kwarg "
                        'must have "$PARAMETER" key'
                    )

        except AssertionError as e:
>           raise InvalidExpectationConfigurationError(str(e))
E           great_expectations.exceptions.exceptions.InvalidExpectationConfigurationError: min_value and max_value cannot both be None

great_expectations/expectations/expectation.py:1121: InvalidExpectationConfigurationError
```